pages/index.tsx

```diff
@@ -1,7 +1,55 @@
+import LandingPage from "@/components/LandingPage";
 import CommunityHome from "@/components/CommunityHome";
+import { GetServerSidePropsContext } from "next";
+import { Community } from "@prisma/client";
+import prisma from "@/lib/prisma";
 
-const Home = () => {
-  return <CommunityHome />;
+const DEFAULT_HOMEPAGE_MATCH = "__NONE__";
+
+const Home = ({
+  communityName,
+  community,
+}: {
+  communityName: string;
+  community: Community;
+}) => {
+  if (communityName === DEFAULT_HOMEPAGE_MATCH) {
+    return <LandingPage />;
+  }
+
+  return <CommunityHome community={community} />;
 };
 
 export default Home;
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  let communityName;
+  let community;
+  const host = context.req.headers.host;
+  const subdomain =
+    process.env.NODE_ENV === "development"
+      ? host?.match(/(.*)\.localhost:3000/)
+      : host?.match(/(.*)\.proplot\.wtf/);
+
+  if (!subdomain) {
+    communityName = DEFAULT_HOMEPAGE_MATCH;
+  } else {
+    communityName = subdomain[1];
+    community = await prisma.community.findFirst({
+      where: {
+        name: communityName,
```

Comment:
I was adding a `uname` property in [my PR](https://github.com/prop-lot/client/pull/24/files#diff-5b443964f4f3a611682db8f7e02177b0a8c632b2039e2bd5e4dd7347815c565cR19) which should match up directly with the subdomain, not sure if we wanna keep that or just go with the name property. I thought the name property may be used as a display name so _could_ differ from the subdomain value (The seed data differs as it's capitalised).

If we wanna keep the uname property i can update this line in my PR to use the uname after this is merged.